crates/scheduler/src/call/crypto/hash.rs

```diff
@@ -81,14 +90,17 @@ fn update<B: Board>(mut call: SchedulerCall<B, api::update::Sig>) {
     let memory = scheduler.applet.store.memory();
     let result = try {
         let data = memory.get(*data, *length)?;
-        match scheduler.applet.hashes.get_mut(*id as usize)? {
+        let context = match scheduler.applet.hashes.take(*id as usize)? {
             #[cfg(feature = "board-api-crypto-sha256")]
-            HashContext::Sha256(context) => context.update(data),
+            HashContext::Sha256(hash) => HashApi(hash).update(data),
```

Comment:
And here you wouldn't need to rewrap, because Sha256 would already contain the wrapped type.
```suggestion
            HashContext::Sha256(context) => context.update(data),
```

Reply:
But we need to call last error right?